app/src/main/java/de/westnordost/streetcomplete/location/LocationStateButton.kt

```diff
@@ -100,13 +101,13 @@ class LocationStateButton @JvmOverloads constructor(
         constructor(superState: Parcelable?) : super(superState)
         constructor(parcel: Parcel) : super(parcel) {
             state = LocationState.valueOf(parcel.readString()!!)
-            activated = parcel.readInt() == 1
+            activated = ParcelCompat.readBoolean(parcel)
         }
 
         override fun writeToParcel(out: Parcel, flags: Int) {
             super.writeToParcel(out, flags)
             out.writeString(state.name)
-            out.writeInt(if (activated) 1 else 0)
+            ParcelCompat.writeBoolean(out, activated)
```

Comment:
I find this rather unnecessary

Reply:
I found it to be an increase in readability (it's nice to have things typed correctly). If you wanted to keep a the same form as the others, you could write extension functions `Parcel.readBooleanCompat()` and `Parcel.writeBooleanCompat()` (if you do this, don't use `readBoolean` and `writeBoolean` as names because I believe extension functions will not override already existing functions — or you could add `Boolean.toInt()` and `Int.toBoolean()` functions instead).

Reply:
Thing is, you don't know (without consulting the documentation) what `readBoolean` will do - there are no boolean types in a parcel - so will it write a string "true" /"false", or an int, short or whatever else? Or will it understand fuzzily everything that is "truthy" (like JS, YAML, PHP,... etc)? The `Boolean.toInt()` approach would be more clear here, but it is such a short statement that I think it is not really worth it.